pkg/controllers/managementuser/secret/project_scoped_secrets.go

```diff
@@ -0,0 +1,209 @@
+}
+
+func (n *namespaceHandler) OnChange(_ string, namespace *corev1.Namespace) (*corev1.Namespace, error) {
+	if namespace == nil || namespace.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	secrets, err := n.getProjectScopedSecretsFromNamespace(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	var errs error
+	desiredSecrets := sets.New[types.NamespacedName]()
+
+	// create/update project scoped secrets
+	for _, secret := range secrets {
+		secretCopy := getNamespacedSecret(secret, namespace.Name)
+
+		s, err := rbac.CreateOrUpdateNamespacedResource(secretCopy, n.secretClient, areSecretsSame)
+		desiredSecrets.Insert(client.ObjectKeyFromObject(s))
+		errs = errors.Join(errs, err)
+	}
+	if errs != nil {
+		return nil, errs
+	}
+
+	return namespace, n.removeUndesiredProjectScopedSecrets(namespace, desiredSecrets)
+}
+
+// getProjectScopedSecretsFromNamespace gets all project scoped secret from a project namespace.
+func (n *namespaceHandler) getProjectScopedSecretsFromNamespace(namespace *corev1.Namespace) ([]*corev1.Secret, error) {
+	// check if namespace is part of a project
+	projectID, ok := namespace.Annotations[projectIDLabel]
+	if !ok {
+		return nil, nil
+	}
+
+	clusterName, projectName, found := strings.Cut(projectID, ":")
+	if !found {
+		logrus.Debugf("Namespace %s projectId annotation %s is malformed, should be <cluster name>:<project name>", namespace.Name, namespace.Annotations[projectIDLabel])
+		return nil, nil
+	}
+
+	p, err := n.projectCache.Get(clusterName, projectName)
+	if err != nil {
+		return nil, fmt.Errorf("error getting project %s for namespace %s: %w", projectName, namespace.Name, err)
+	}
+
+	backingNamespace := p.GetProjectBackingNamespace()
+
+	r, err := labels.NewRequirement(projectScopedSecretLabel, selection.Equals, []string{projectName})
+	if err != nil {
+		return nil, err
+	}
+	return n.managementSecretCache.List(backingNamespace, labels.NewSelector().Add(*r))
+}
+
+// removeUndesiredProjectScopedSecrets removes project scoped secrets from the namespace that are not in the desiredSecrets map.
+func (n *namespaceHandler) removeUndesiredProjectScopedSecrets(namespace *corev1.Namespace, desiredSecrets sets.Set[types.NamespacedName]) error {
+	// remove any project scoped secrets that don't belong in the project
+	downstreamProjectScopedSecrets, err := n.secretClient.List(namespace.Name, metav1.ListOptions{
+		LabelSelector: projectScopedSecretLabel,
+	})
+	if err != nil {
+		return err
+	}
+
+	allSecrets := sets.New[types.NamespacedName]()
+	for _, secret := range downstreamProjectScopedSecrets.Items {
+		allSecrets.Insert(client.ObjectKeyFromObject(&secret))
+	}
+
+	secretsToDelete := allSecrets.Difference(desiredSecrets)
+
+	var errs error
+	for _, secret := range secretsToDelete.UnsortedList() {
+		// secret in namespace does not belong here
+		logrus.Infof("deleting secret %s from namespace %s", secret.Name, secret.Namespace)
```

Comment:
I believe we make log messages begin with a capital...but, also, maybe this could provide _more context_ as to why it's deleting a secret?

"Cleaning project scoped secret %s from namespace %s" or something?